【Hackathon No.1&No.2】为 Paddle 新增 finfo API和iinfo API

isLinXu opened the pull request:
<!-- Demo: https://github.com/PaddlePaddle/Paddle/pull/24810 -->
### PR types
<!-- One of [ New features | Bug fixes | Function optimization | Performance optimization | Breaking changes | Others ] -->
New features
### PR changes
<!-- One of [ OPs | APIs | Docs | Others ] -->
APIs
### Describe
<!-- Describe what this PR does -->
为 Paddle 新增 finfo API和iinfo API
1.add type_info.cc&.h
2.add tinfo.cc&.h

Commit:
1.add API for paddle.finfo
2.add API for paddle.iinfo
pr url(https://github.com/PaddlePaddle/community/pull/83)
issue:#40333;40334

isLinXu committed:
Merge branch 'PaddlePaddle:develop' into develop

isLinXu committed:
Merge branch 'PaddlePaddle:develop' into develop

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/PaddlePaddle/Paddle?pullRequest=42147) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/PaddlePaddle/Paddle?pullRequest=42147) before we can accept your contribution.<br/><hr/>**isLinXu** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/PaddlePaddle/Paddle?pullRequest=42147) it.</sub>

Commit:
1.add API for paddle.finfo
2.add API for paddle.iinfo
pr url(https://github.com/PaddlePaddle/community/pull/83)
issue:#40333;40334

isLinXu committed:
Merge branch 'PaddlePaddle:develop' into develop

isLinXu committed:
Merge branch 'PaddlePaddle:develop' into develop

Commit:
Merge branch 'develop' of github.com:isLinXu/Paddle into develop

Commit:
1.add API for paddle.finfo
2.add API for paddle.iinfo
pr url(https://github.com/PaddlePaddle/community/pull/83)
issue:#40333;40334
